object-storage-aws/src/main/java/io/micronaut/objectstorage/aws/AwsS3ObjectStorageEntry.java

```diff
@@ -58,4 +61,16 @@ public InputStream getInputStream() {
     public GetObjectResponse getNativeEntry() {
         return responseInputStream.response();
     }
+
+    @NonNull
+    @Override
+    public Map<String, String> getMetadata() {
+        return responseInputStream.response().metadata();
```

Comment:
Is the map returned here immutable? If not wrap in `Collections.unmodifiableMap`